src/iop/colorin.c

```diff
@@ -875,6 +875,39 @@ static void process_cmatrix_fastpath_clipping(struct dt_iop_module_t *self, dt_d
   }
 }
 
+static void rgb_to_uvY(float rgb[3], float uvY[3])
+{
+  float X = 2.7689f * rgb[0]
```

Comment:
where do the RGB -> XYZ come from ?

Reply:
That's wikipedia coefficients for cieRGB to cieXYZ conversion

Reply:
https://fr.wikipedia.org/wiki/CIE_XYZ#Passage_de_l'espace_CIE_XYZ_%C3%A0_l'espace_CIE_RGB

Was it not the right space to use?

Reply:
But why CIE RGB ? It's either camera RGB or pipeline RGB we need, depending on where you do the mapping, or even XYZ directly. 

XYZ is obtained from camera RGB by a matrice dot product with a matrice produced by least-squares fitting RGB -> XYZ over known calibration samples (a color checker or an IT8 chart), which is essentially the Adobe DNG matrices.

The problem of this method is it is designed for accurate mapping in the central spectrum region, but doesn't care about gamut bounds, so the transvection can result in bad coordinates.

Reply:
I agree that before colorin, we are in camera rgb, after colorin, either Lab or the RGB working space.

I am in colorin, right after the matrice product.
In colorin, we do a matrice product with camera RGB data, but even if it is written xyz everywhere in the variable names, I believe the output data is RGB, not XYZ:
if I do _xyz[1] /= 4; for all pixels, we see an effect which looks like a scaling on the green channel, not a change of luminance:
see on this image:
![Capture d’écran du 2020-05-25 15-18-41](https://user-images.githubusercontent.com/34063828/82816468-7b103280-9e9b-11ea-924d-c2d36a4a7755.png)

doing _xyz[1] /= 4; for all pixels right after the matrice product (see how the histogramme has not changed for R and B, but has changed for G):
![Capture d’écran du 2020-05-25 15-20-59](https://user-images.githubusercontent.com/34063828/82816471-7ea3b980-9e9b-11ea-9226-bda3014b423e.png)

As I could not take the RGB working space, nor the camera rgb space for the XYZ conversion, I assumed that we were in cieRGB, but maybe that's not the case

Reply:
What you see is not what you get. Even in XYZ… Y is both luminance and "green" stimulus, this is anchored in perception. For practical ends, XYZ is yet another RGB space, only it is a reference one that is gluing spectral input to physiological output. But after the dot product, you should only care about XYZ.

Reply:
Aaaah ok I understand.
Thanks for taking time to explain that :-)